flink-formats/flink-avro/src/test/java/org/apache/flink/formats/avro/AvroExternalJarProgramITCase.java

```diff
@@ -44,49 +39,24 @@
 
 	private static final String TEST_DATA_FILE = "/testdata.avro";
 
-	@Test
-	public void testExternalProgram() {
-
-		LocalFlinkMiniCluster testMiniCluster = null;
+	private static final MiniClusterResource miniClusterResource = new MiniClusterResource(
+		new MiniClusterResource.MiniClusterResourceConfiguration(
+			new Configuration(),
+			1,
+			4));
 
+	@Test
+	public void testExternalProgram() throws Exception {
+		miniClusterResource.setJarFiles(Collections.singleton(new Path(JAR_FILE)));
```

Comment:
I think instead of passing the `JAR_FILE` to the `MiniClusterResource` we should set the `TestEnvironment` with the correct jar files settings explicitly in this test.

Reply:
Btw: Do we actually need to explicitly add the `JAR_FILE`? Doesn't the `PackagedProgram` take care of this for us?

Reply:
as far as i could tell, no. We actually have to set it in the environment.

Reply:
Ok, maybe we could verify this.

Reply:
The jar passed to the `PackagedProgram` is only used to create the classloader for loading the contained classes.

The jars have to be explicitly added to the JobGraph, which we here do by passing it to the environment that does it for us. In the CLI, we use [PackagedProgramUtils](https://github.com/apache/flink/blob/master/flink-clients/src/main/java/org/apache/flink/client/program/PackagedProgramUtils.java) to retrieve the jars from the PackagedProgram and register them on the jobgraph.